### PR TITLE
On demand

### PR DESCRIPTION
**Pull request by amiturgman**
@morsh please review

### CODE REVIEW

**Comment by amiturgman** [comment]
@morsh - please review. let me know if you have no comments/questions so I can merge.

File: bot/handlers/set_alarm.js [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
-  var intent = session.dialogData[data.source];
+  var intent = session.dialogData.data[data.source];
   var alarmTime = null;
   if (intent.actions[0].parameters[0].name == "time") {
+    // alarmTime = intent.entities...
+    // use intent.entities to extract relevant information
+    // assuming extracted alarmTime 
 
-    alarmTime = intent.actions[0].parameters[0].value[0].resolution.time;
+    alarmTime = '2016-10-10 10:10';
```
Comment (morsh) [inline bot/handlers/set_alarm.js:11]: why is this constant?

**Review by morsh — COMMENTED** [review]
@amiturgman 
one main concern - index and loadondemand are very complex and hard to follow (even for me, and I know the code)
I think that the main file should be routers.js and maybe some of the logic in index and load should be in the bot-graph-dialog enable a simple call instead of a complex implementation.

If someone wants to change something different, they can follow a more complex scenario.
